tools/wasi/wasi.cpp

```diff
@@ -152,6 +152,13 @@ bool run(bytes_view wasm_binary, int argc, const char* argv[], std::ostream& err
         {ns, "environ_get", {ValType::i32, ValType::i32}, ValType::i32, return_enosys},
     };
 
+    auto module = parse(wasm_binary);
+    auto imports = resolve_imported_functions(*module, wasi_functions);
+    return instantiate(std::move(module), std::move(imports), {}, {}, {}, MaxMemoryPagesLimit);
+}
```

Comment:
We had an `assert(instance != nullptr)` in the old place. Worth keeping here or `unique_ptr` ensures it?

Reply:
`unique_ptr` doesn't care, but `instantiate` reports errors by throwing, so I don't think it's useful.